### ISSUE TITLE
SchemaGenerator parameters of type list

### ISSUE THREAD

**Issue by DariaKnyazeva** [tomchristie/django-rest-framework#4485]
Hi,

I have a custom method in ViewSet with @list_route decorator. It expects a list of dictionaries in request.POST. How is it possible to generate coreapi Document with parameters (fields) of type "array" or "list"?

**Comment by tomchristie** [comment]
> It expects a list of dictionaries in request.POST.

Could you expand on that? What part of the system or interface "expects" a list of dictionaries. Are we referring to a Swagger UI interface, the coreapi client, or something else, and can you give an example?

The parameters that coreapi exposes are untyped - it's up to the server to validate the input (eg. using a serializer)

**Comment by DariaKnyazeva** [comment]
An example of method I would like to get swagger interface work:
```
class MyViewSet(mixins.CreateModelMixin,
                            mixins.RetrieveModelMixin,
                            mixins.ListModelMixin,
                            viewsets.GenericViewSet):
    @list_route(methods=['post'])
    def pricing(self, request):
       """
       Expects the following request structure:
            [{ 'start_date': value,
                'end_date': value,
                'code': value,},
             {start_date, end_date, code}, {start_date, end_date, code}, ... ]
        """
       for request_dict in request.data:
            do something 
```

My question is, I would like to have the required fields in 'Try it out!' swagger form: a group of 'start_date', 'end_date' and 'code' parameters with the possibility to add / remove the groups.

**Comment by tomchristie** [comment]
I'm going to backlog this for 3.5. Really this is an issue of us properly documenting how to customize the schema endpoints, which still needs some improvement.

**Comment by daimon99** [comment]
I encounter the same problem too. Hope  there is a good solution. Now I hava to write my `get_link()`

**Comment by tomchristie** [comment]
> I would like to have the required fields in 'Try it out!' swagger form: a group of 'start_date', 'end_date' and 'code' parameters with the possibility to add / remove the groups.

Gothca, understand the issue now! Yes, this has been an issue in how we generate the schema for swagger UI. It's resolved as part of the upcoming 3.5 release.

**Comment by ramusus** [comment]
Where I can read how to generate the schema?
